src/components/lib/sl/sl.c

```diff
@@ -343,20 +411,36 @@ sl_thd_alloc(cos_thd_fn_t fn, void *data)
 	struct sl_thd *t;
 
 	sl_cs_enter();
-	t = sl_thd_alloc_intern(fn, data, NULL, 0);
+	t = sl_thd_alloc_intern(fn, data);
 	sl_cs_exit();
 
 	return t;
 }
 
-/* Allocate a thread that executes in the specified component */
 struct sl_thd *
-sl_thd_comp_alloc(struct cos_defcompinfo *comp)
+sl_thd_aep_alloc(cos_aepthd_fn_t fn, void *data, int owntc)
```

Comment:
I like changing the `tc` to `tcap` and perhaps two words `own_tcap`, or better `separate_tcap`.  I don't think that everyone will know what a `tc` is as it is an edge of the api.